src/oidc_bridge.rs

```diff
@@ -78,14 +75,16 @@ pub fn request(app: &Config, email_addr: EmailAddress, client_id: &str, nonce: &
         "?",
         "client_id=",
         &utf8_percent_encode(&provider.client_id, QUERY_ENCODE_SET).to_string(),
-        "&response_type=code",
+        "&response_type=id_token",
         "&scope=",
         &utf8_percent_encode("openid email", QUERY_ENCODE_SET).to_string(),
         "&redirect_uri=",
         &utf8_percent_encode(&format!("{}/callback", &app.public_url),
                              QUERY_ENCODE_SET).to_string(),
         "&state=",
         &utf8_percent_encode(&session, QUERY_ENCODE_SET).to_string(),
+        "&nonce=",
+        &utf8_percent_encode(&nonce, QUERY_ENCODE_SET).to_string(),
```

Comment:
This reminds me: The spec mentions in https://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthRequest specifically for the implicit flow that the nonce needs to be unguessable. I guess with the client having access to the token that gets more important.

But If I see https://github.com/portier/portier-broker/blob/b72bcd195ed1503512ce662a085624bf6a4cbcef/src/handlers/oidc.rs#L120 right, we are using the nonce we get in the /auth request directly from the RP (I absolutely might be wrong here!). But if that is the case, we can't actually trust that nonce. We would need to create a nonce specifically for our own OIDC flow with the IdP.

We are probably already doing that and I'm just missing it in the Rust code?